### ISSUE TITLE
Card ID has a mistake

### ISSUE THREAD

**Issue by idotj** [GeorgeYT9769/cardabase-app#4]
Hi, first of all, thanks for this nice app!
It works as expected but I found a bug that I fixed manually just typing a number missing in the barcode.

Here you have the barcode from a fidelity card in a supermarket where I found the bug:

![image](https://github.com/user-attachments/assets/eb668f70-6877-4db7-aac6-0667da640ea3)

At the moment I click on the camera icon to add the Card numbers, it takes/detects the barcode but omitted the first number (as you can see in the picture it's a zero "0").
When I click on the button "Save" I get the message: Card ID has a mistake

Luckily I was able to check that the problem is that the first number does not appear, and by entering it manually the problem is solved.

I'm using the version 1.0.0 from F-Droid in a smartphone running Android 13 

FYI, maybe this bug is related with issue #2

**Comment by GeorgeYT9769** [comment]
Hi, thanks for using the app. The bug that you discovered is related to package that I am using for the app to scan barcodes. So the problem might be there. Does this happen on other cards too? I've never experienced something like that, it always worked fine on my side...

**Comment by GeorgeYT9769** [comment]
Oh okay, I will add an exception for that. Thanks.

**Comment by GeorgeYT9769** [comment]
So what I found is that many more apps have this issue. They are released on Google Play Store and probably use better packages, but still don't take the 0.
Adding an exception could work, but later when I'll allow more types of codes, it could be a problem.
So I'll leave it as open and see what I can do about it.

**Comment by aplufr** [comment]
Hello, 
I do have the same issue with a card with 19 digits. Binary eyes report it as an "19 chars, CODE 128"

**Comment by GeorgeYT9769** [comment]
Hello @aplufr, other types of barcodes will be added soon. I am currently working on it.

**Comment by thomaspaulb** [comment]
I have a 14 digit barcode that's not working either! Is there anything i can do to help support it?

**Comment by GeorgeYT9769** [comment]
Hello, and sorry for my late response, didn't see the notification for a new comment... As I mentioned earlier, for now only EAN-13 card types are supported. (Right now in the private dev build I've added like 10 other types of cards, but the cards added before the addition are causing bugs 😑). Also what can you offer? Any help would be appreciated... 🙂> I have a 14 digit barcode that's not working either! Is there anything i can do to help support it?

**Comment by thomaspaulb** [comment]
Ok your response is within the hour so I think you're good as response time goes :-D

I'm a programmer, though mostly in Python and Javascript, I've done some Flutter. I guess what I meant is, I can try my luck at patch to support N-number barcodes instead of only 13 digits, or, I can test your dev build and try to find the problem with it. I'm usually pretty good at finding bugs. But... the dev build is not public?

**Comment by GeorgeYT9769** [comment]
Hello, I read 1 month and wonderer why the email came so late 🤣. No, the dev build is not public, it's private for my testing purposes. Contact me on Discord "georgeyt9769" and we will sort something out (all about the current situation, bugs...)

**Comment by thomaspaulb** [comment]
Nice! I'm going to test this out on my strange barcode. I didn't get to Discord you yet, busy as it is... will report result.

**Comment by GeorgeYT9769** [comment]
That's fine, I was thinking about making a server on Discord for anyone, because discussing things via issues isn't the best or the fastest

**Comment by thomaspaulb** [comment]
Personally I would suggest just pushing the dev branch here AND doing everything via issues, but I'm just old fashioned :) you could also open a snapchat and an IG page hehe

**Comment by GeorgeYT9769** [comment]
Could do that, but why would I make it harder lol. Also I don't have a dev branch 😆 (good idea, might make one)

**Comment by thomaspaulb** [comment]
You do you, at least you've made a cool open source app that I can now use at the local supermarket, it saves me from blushing and explaining I've forgotten the card AGAIN and if they would kindly look me up in the system and give me the discount

**Comment by GeorgeYT9769** [comment]
Yeah, thanks so much! Happy to see someone actually enjoying it. And the dev branch seems like a good option for code backup and sharing the things that are still in the development.
